futurenhs.app/components/NotificationBanner/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { useRef, useState, useEffect } from 'react'
+import classNames from 'classnames'
+
+import { Props } from './interfaces'
+import { RichText } from '@components/RichText'
+import { notifications } from '@constants/notifications'
+
+export const NotificationBanner: (props: Props) => JSX.Element = ({
+    text,
+    className,
+}) => {
+
+    const wrapperRef: any = useRef(null)
+
+    const { heading, main } = text ?? {}
+    const notificationHeading: string = heading ? heading : notifications.SUCCESS
+
+    const isAlert: boolean = notificationHeading === notifications.SUCCESS
+
+    const generatedClasses: any = {
+        wrapper: classNames('govuk-notification-banner', {['govuk-notification-banner--success']: isAlert}, 'u-mt-6', className),
+        header: classNames('govuk-notification-banner__header'),
+        title: classNames('govuk-notification-banner__title'),
+        body: classNames('govuk-notification-banner__content'),
+        bodyHeading: classNames('govuk-notification-banner__heading')
+    }
+
+    useEffect(() => {
+
+        const wrapper: HTMLElement = wrapperRef?.current
+        wrapper?.setAttribute('tabIndex', '-1')
+        wrapper?.classList?.add('focus:u-outline-none')
+        wrapper?.focus()
+        wrapper?.addEventListener('blur', () => {
+            wrapper?.removeAttribute('tabIndex')
+        })
+
+    }, [wrapperRef.current])
+
+    return (
+        <div
+            className={generatedClasses.wrapper}
+            role={isAlert ? 'alert' : 'region'}
+            aria-labelledby="notification-banner-title"
+            ref={wrapperRef}
+        >
+            <div className={generatedClasses.header}>
+                <h2 className={generatedClasses.title} id="notification-banner-title">{notificationHeading}</h2>
```

Comment:
Hardcoded IDs in components can be problematic - you have no control over avoiding duplicate IDs occurring at page level if more than one instance of the component is rendered - unlikely in this case but not impossible.

Recommend a required id prop for this component which you can then generate an id for this e.g.

const generatedIds: any = {
    title: `${id}-botification-banner-title`
}